lib/rubocop/ext/regexp_node.rb

```diff
@@ -16,12 +16,16 @@ class << self
       @parsed_cache = {}
 
       # @return [Regexp::Expression::Root, nil]
-      def parsed_tree
-        return if interpolation?
+      def parsed_tree(interpolation: :ignore)
+        unless %i[ignore blank].include?(interpolation)
```

Comment:
When thinking about the fact that `:ignore` is confusing (it's "ignore the regexp itself", not the interpolations), the only good name I could come up with was `:return_nil` and that's a code smell.

That made me wonder if the best course of action isn't to remove the option altogether, and make blanking the default. What do you think? It should be the callers job to check for interpolation  `lint/mixed_regexp_capture_types` would need to do that).

Should be documented, and marked as breaking change in the Changelog.

Reply:
I agree, I'm not really a fan of `interpolation :ignore`. I like the idea of always blanking (that is, until we find a scenario where it isn't appropriate 😄). 

Having had a quick look at `lint/mixed_regexp_capture_types` I think it could also just use blanking, as we could catch strictly more issues (where the regexp contains mixed capture groups in addition to an interpolation)

I'll take a look at making this change, and report back

Reply:
Oh, you're right about `lint/mixed_regexp_capture_types` 🤦‍♂️

Reply:
So for now, to keep this PR focussed, I left `lint/mixed_regexp_capture_types` as is - i.e. it still does nothing in the presence of interpolation. A nice follow-up might be to change it, as I mentioned previously